lib/adagrams.rb

```diff
@@ -0,0 +1,88 @@
+require "csv"
+
+def draw_letters()
+  letter_pool = ["A"] * 9 + ["B"] * 2 + ["C"] * 2 + ["D"] * 4 + ["E"] * 12 + ["F"] * 2 + ["G"] * 3 + ["H"] * 2 + ["I"] * 9 + ["J"] + ["K"] + ["L"] * 4 + ["M"] * 2 + ["N"] * 6 + ["O"] * 8 + ["P"] * 2 + ["Q"] + ["R"] * 6 + ["S"] * 4 + ["T"] * 6 + ["U"] * 4 + ["V"] * 2 + ["W"] * 2 + ["X"] + ["Y"] * 2 + ["Z"]
+  hand = []
+
+  # Shuffling once resulted in too many duplicates.
+  # The program shuffles multiple times.
+  10.times do
+    letter_pool.shuffle!
+    hand << letter_pool.pop
+  end
+
+  return hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  split = input.upcase.split("")
+  copy = letters_in_hand.dup
+
+  split.each do |letter|
+    if copy.include? (letter)
+      copy.delete_at(copy.index(letter))
+    else
+      return false
+    end
+  end
+
+  return true
+end
+
+def score_word(word)
+  word_arr = word.upcase.split("")
+  points = 0
+  word_arr.each do |letter|
```

Comment:
While this `case` statement works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}
```

Then to get the score for a letter, you can say `LETTER_SCORES[letter]`.